tests/phpunit/tests/abilities-api/wpAbilitiesRegistry.php

```diff
@@ -651,3 +675,32 @@ static function ( $args, $name ) {
 		$this->assertNotSame( $filtered_ability->get_label(), $unfiltered_ability->get_label(), 'The filter incorrectly modified the args for an ability it should not have.' );
 	}
 }
+
+/**
+ * Test custom ability class that extends WP_Ability.
+ *
+ * This class overrides do_execute() and check_permissions() directly,
+ * allowing registration without execute_callback or permission_callback.
+ */
+class Tests_Custom_Ability_Class extends WP_Ability {
```

Comment:
This should be move outside of the test file, to `tests/phpunit/includes`

Reply:
Done! d8a71e5e8b093e359fde031d9e1ad1ff7c2eb9c7